pytensor/tensor/basic.py

```diff
@@ -3726,109 +3727,19 @@ def trace(a, offset=0, axis1=0, axis2=1):
     return diagonal(a, offset=offset, axis1=axis1, axis2=axis2).sum(-1)
 
 
-class AllocDiag(Op):
-    """An `Op` that copies a vector to the diagonal of a zero-ed matrix."""
+class AllocDiag(OpFromGraph):
+    """
+    Wrapper Op for alloc_diag graphs
+    """
 
     __props__ = ("offset", "axis1", "axis2")
```

Comment:
I guess offset was not allowed to be symbolic before, do we want that to be the case? Otherwise, if symbolic works, we can add it as the second output so it's still easy to retrieve and check

Reply:
as an output to the function? wouldn't that be confusing for users

Reply:
Sorry as a second input. Typo

Reply:
Sure but that means offset cannot be symbolic. Checking if the last output is a constant zero isn't much harder, you can even add a helper method to the op

```python
def is_offset_zero(self, node) -> bool:
  offset = node.inputs[-1]
  return isinstance(offset, Constant) and offset.data.item() == 0
```

Then you can replace checking `op.offset==0` by `op.is_offset_zero(node)`. 

This assuming the inner graph can actually be written with symbolic offsets, which I think it can.